Port to python3 for next major Mopidy release

Gadgetoid opened the pull request:
Aims to fix https://github.com/pimoroni/mopidy-raspberry-gpio/issues/1

Gadgetoid committed:
Update project to match cookiecutter-mopidy-ext

Gadgetoid committed:
pyupgrade --py37-plus **/*.py

Gadgetoid committed:
black .

Gadgetoid committed:
isort -rc . && black .

Gadgetoid committed:
Use pathlib in Extension

Gadgetoid committed:
Use unittest.mock instead of mock

Gadgetoid committed:
Use f-strings instead of % and .format()

Gadgetoid committed:
Fix pinconfig encode,decode and linting

Gadgetoid committed:
dos2unix crlf -> lf

Gadgetoid committed:
Fix silly blunder

Gadgetoid committed:
Expand frontend test coverage

Gadgetoid committed:
Remove cookiecutter raw markup

jodal commented:
To fix the build error here, you need to:

- Go to CircleCI organization security settings, e.g. https://circleci.com/gh/organizations/pimoroni/settings#security
- Check "Yes, allow all members of my organization to (...) use third-party orbs"

You only need to do this for the first repo in your org you use with CircleCI.

Gadgetoid committed:
Poke CircleCI to recompile config

Gadgetoid committed:
Fix for 3.x API changes

Switches from use of playback controller (deprecated) to mixer volume as mentioned in #1

Uses `get_state()` instead of `state`

Gadgetoid committed:
Apply black reformatting suggestions

Gadgetoid committed:
Supplied dummy Mopidy Core to frontend for testing

This currently uses three files copy-n-pasted from upstream Mopidy to provide dummy interfaces for mixer, backend and audio.

These dummy files seem a little janky- should they be part of a mopidy-test package so that they stay in-sync with Mopidy's API? Since otherwise tests may?? pass that fail against the real Mopidy. Maybe?

Gadgetoid committed:
Black/Flake8 linting fixes

Gadgetoid committed:
Fix tests hanging
